### ISSUE TITLE
Generalized Advantage Estimation / GAE ?

### ISSUE THREAD

**Issue by BigBadBurrow** [nikhilbarhate99/PPO-PyTorch#9]
Do you have any knowledge of GAE to possibly implement as an extra PPO example?

**Comment by nikhilbarhate99** [comment]
Although I have read about it, the code gets quite complicated for a minimal implementation. I don't plan to add it anytime soon.

If you want a intuitive explanation of GAE refer to this [link](https://lilianweng.github.io/lil-log/2018/02/19/a-long-peek-into-reinforcement-learning.html#value-estimation)

**Comment by BigBadBurrow** [comment]
That's a great webpage, thanks for sharing that - one for my bookmarks.  I found a nice, relatively simple implementation of PPO using GAE, but downside is it's in Tensorflow and I need to use PyTorch. Perhaps between us we can port it over?

https://github.com/bsivanantham/GAE/blob/master/ppo/ppo.py

**Comment by nikhilbarhate99** [comment]
Like I said. I don't plan to add it anytime soon.

You can give it a try if you want to ..... anyways PyTorch or Tensorflow does not make any difference since it does not affect the graph, you could copy the code for `GAE` and `discount function` from the repo in the link and modify it. Note that the `discount function` also uses a linear filter from scipy.
